### DISCUSSION TITLE
Help with this PET module error message

### DISCUSSION THREAD

**Discussion by Tequila0211** [CenterForTheBuiltEnvironment/pythermalcomfort#66] [Q&A] [answered]
Problems with the Pet module, when I run my model with my Dataframe I get this response: 
" File "C:\UsersChristian\AppData\Local\Programs\Python\Python310\libsite-packages\pythermalcomfort\models.py", line 3066, in solve_pet
    w = esw / emax # skin wettedness
ZeroDivisionError: float division by zero
PS C:\Users ChristianDesktopColombiaResarchCodeZoneZoneClimateColombia> " 

However with other modules like UTCI or PMV I get no problems with the same data.

**Comment by FedericoTartarini** [comment]
Very good point, a colleague of mine just raised the same issue. I will work on it and try to implement a solution in the coming days. Could you please share the code that is generating the error?

**Reply by Tequila0211** [reply]
I want to add a curiosity that this error does not occur with all the analyzed data, only with some. However, I have not been able to identify the error pattern because the failure has occurred randomly and in different weather configurations. 

 There is an example of the code. It is important to mention that it is analyzing TMY data for multiple points.

`import pandas as pd
import os
import numpy as np
import thermofeel
import pythermalcomfort.models as tcmodels 
import pythermalcomfort.psychrometrics as tcpsy
import multiprocessing as mp

PEDEANTRANSMET = 2.0
def clotingInsulationByTemperature(elevation):
    condition_1 = elevation <= 1000
    condition_2 = (elevation > 1000) & (elevation <= 2000)
    condition_3 = (elevation > 2000) & (elevation <= 3000)
    clo = np.where(condition_1, 0.39, np.where(condition_2, 0.47, np.where(condition_3, 0.91, 1.25)))
    return clo

# Ruta de la carpeta que contiene los archivos CSV
folderRoot = r"C:\Users\cristian\Desktop\ColombiaResarchCode\citiesForEvaluate\Bogota"

def ThermalAnalysis(root,file):
    
    location = pd.read_csv(os.path.join(root, file))
    elevation = int(location.iloc[0,8])
    data = pd.read_csv(os.path.join(root, file),skiprows=2)
    data.loc[:,'Year'] = 2021
    data['Date'] = pd.to_datetime(data[['Year', 'Month', 'Day', 'Hour','Minute']])
    
    temperature = data['Temperature']
    dewPoint = data['Dew Point']
    windSpeed = data['Wind Speed']
    conditionW = windSpeed <= 0
    windSpeedNew = np.where(conditionW,0.1,windSpeed)
    data['ModWindSpeed'] = windSpeedNew
    modWS = data['ModWindSpeed']
    modWSn = modWS.to_numpy()
    
    pressure = data['Pressure']
    condition = data['GHI'] > 0
    tgiv2 = (1.33*data['Temperature'] -2.65*np.sqrt(data['Temperature'])+3.21*np.log(data['GHI']+1)+3.5)
    tgi = (1.36 * data['Temperature'] - 2.358) * (0.0747 * np.log(data['GHI']) + 0.5617)
    tgv2 = np.where(condition, tgiv2, 0.942 * data['Temperature'])
    tg = np.where(condition, tgi, 0.942 * data['Temperature'])
    data['tgv2'] = tgv2
    data['tg'] = tg
    data['clo'] = clotingInsulationByTemperature(elevation)
    clo = data['clo']
    clon = clo.to_numpy()
    mrt1 = (1+0.22*(data['Wind Speed']**0.5))*(data['tg']-data['Temperature'])+data['Temperature']
    mrt2 = (1+0.22*(data['Wind Speed']**0.5))*(data['tgv2']-data['Temperature'])+data['Temperature']
    data['mrt1'] = mrt1
    data['mrt2'] = mrt2
    globetemperature = data['tg']
    
    globetemperaturen = globetemperature.to_numpy()
    temperaturen = temperature.to_numpy()
    dewPointn = dewPoint.to_numpy()
    windSpeedn = windSpeed.to_numpy()
    pressuren = pressure.to_numpy()
    
    ktdb = temperaturen +273.15
    ktd = dewPointn +273.15
    
    rh = thermofeel.calculate_relative_humidity_percent(ktdb,ktd)
    tmrt = tcpsy.t_mrt(globetemperaturen,temperaturen,windSpeedn)
    
    def petcal (temperature,tmrt,ws,rh,clo,pressure):
       return tcmodels.pet_steady(temperature,tmrt,ws,rh,PEDEANTRANSMET,clo,pressure,position=2,age=35,sex=1,weight=75,height=1.75,wme=0)
    petvector = np.vectorize(petcal)
    pet = petvector(temperaturen,tmrt,modWSn,rh,clon,pressuren)
    
    data['PET'] = pet
    FinalData = data [['Date','Hour','Temperature','Dew Point','Wind Speed','Relative Humidity','GHI','tg','tgv2','Mean Radiant Temperature','mrt1','mrt2','PET']]
    
    # Nombre de la columna que contiene la fecha y hora
    columna_fecha_hora = 'Date'
    columnaHora = 'Hour'

    initial5Hours = FinalData.iloc[:5]
    # Almacena la columna de fecha y hora en una variable
    fecha_hora = FinalData[columna_fecha_hora]
    hour2 = FinalData[columnaHora]

    # Desplaza todas las columnas excepto la de fecha y hora en 5 puestos hacia arriba
    FinalData = FinalData.iloc[:, 1:].shift(-5, axis=0)
    FinalData.iloc[-5:] = initial5Hours
    # Restaura la columna de fecha y hora
    FinalData[columna_fecha_hora] = fecha_hora
    FinalData[columnaHora] = hour2
    
    filename = f'V3HourlyEvaluation_{file}.csv'
    
    FinalData.to_csv(filename, index=False)
    
    return FinalData

def process_file(folder,file):
    example =  ThermalAnalysis(folder,file)
    
    return example
    

def bigAnalysis(folderToAnalyze):
    
    
    files_csv = os.listdir(folderToAnalyze)
    pool = mp.Pool(5)
   
    for file in files_csv:
        evaluation = pool.apply_async(process_file, args=(folderToAnalyze, file))
        
    pool.close()
    pool.join()
    return evaluation


if __name__ == '__main__':
    bigAnalysis(folderRoot)
`

**Reply by FedericoTartarini** [reply]
The error is most likely caused by the fact that the skin temperature and relative humidity are the same as the environmental ones. Consequently, the amount of heat that can be dissipated via sweating is 0. 

The code is trying to calculate skin wettedness by dividing what is required for thermal balance over how much it can be dissipated. However, since the latter is 0 an error is raised since a division by zero cannot be done. We have already identified a solution I just need to finalise the code and implement it.

When do you need this done by?

**Reply by Tequila0211** [reply]
Hi federico I'm glad you found the error, thank you for your introduction to this problem and its solution. Since I'm processing other thermal indexes besides this one I might continue with them in the meantime.

Maybe in a week I will need the solution to continue my research, however, I do not know the time you require for this implementation so I am attentive to your comments.

 I thank you for the prompt attention you have given to this issue.

**Reply by FedericoTartarini** [reply, marked as answer]
Hi, I have just fixed the issue and released a new version of pythermalcomfort. Could you please download it and check if the issue is gone?

**Reply by Tequila0211** [reply]
Hi Federico,
I am informing you that I have already made some preliminary tests, and your implementation of the PET index is working well. 
However, if I detect anything rare, I will let you know. 
Thank you very much for your collaboration.

**Reply by FedericoTartarini** [reply]
I am glad it worked. Yes, please message me or open an issue here on GitHub if you notice that something is not working well.
